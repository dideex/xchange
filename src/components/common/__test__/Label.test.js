import React from 'react'
import Component from '../Label'
import {shallow, mount} from 'enzyme'

describe('Label behaviour', () => {
  describe('Markup', () => {
    it('Base markup', () => {
      const snapshotWrapper = shallow(<Component caption="test label" />)
      expect(snapshotWrapper.html()).toMatchSnapshot()
    })

    it('Base with custom styles', () => {
      const snapshotWrapper = shallow(
        <Component caption="test label" style={{test: 'new test styles'}} />,
      )
      expect(snapshotWrapper.html()).toMatchSnapshot()
    })

    it('Base with icon', () => {
      const snapshotWrapper = shallow(<Component caption="test label" icon="Bitcoin" />)
      expect(snapshotWrapper.html()).toMatchSnapshot()
    })

    it('Base with big icon', () => {
      const snapshotWrapper = shallow(<Component caption="test label" big />)
      expect(snapshotWrapper.html()).toMatchSnapshot()
    })

    it('Base with big draggin icon', () => {
      const snapshotWrapper = shallow(<Component caption="test label" big isDragging />)
      expect(snapshotWrapper.html()).toMatchSnapshot()
    })
  })

  it('Custom onclick handler', () => {
    const click = jest.fn()
    const wrapper = mount(<Component caption="test label" onClick={click}/>)
    
    wrapper.simulate('click')
    expect(click).toHaveBeenCalledTimes(1)
  })
})
