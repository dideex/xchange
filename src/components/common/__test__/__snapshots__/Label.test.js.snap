// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label behaviour Markup Base markup 1`] = `"<span style=\\"color:#191716;background:transparent\\" class=\\"css-t2xfrr\\"><span style=\\"padding-left:45px\\">test label</span></span>"`;

exports[`Label behaviour Markup Base with big draggin icon 1`] = `
"<span style=\\"color:#fff;background:#aee239\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity:0\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill:#aee239\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill:#8fbe00\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><span style=\\"padding-left:0\\">test label</span></span>"
`;

exports[`Label behaviour Markup Base with big icon 1`] = `
"<span style=\\"color:#191716;background:transparent\\" class=\\"css-t2xfrr\\"><svg style=\\"opacity:1\\" class=\\"currency_bg\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 600 600\\"><path style=\\"fill:#aee239\\" d=\\"M600,300c0,5.8-0.2,11.5-0.5,17.2c-0.4,7.6-1.1,15.2-2.1,22.6c-5.5,41.1-19.2,79.6-39.6,113.7
c-49.6,83.2-138.3,140.3-240.8,146c-5.6,0.3-11.3,0.5-17,0.5C134.3,600,0,465.7,0,300S134.3,0,300,0S600,134.3,600,300z\\"></path><path style=\\"fill:#8fbe00\\" d=\\"M443,160.7c-36.2-36.9-86.6-59.8-142.4-59.8c-110.2,0-199.5,89.3-199.5,199.5c0,54.4,21.8,103.8,57.2,139.8l0,0
l0.6,0.6c0.4,0.4,0.9,0.9,1.3,1.3l157,157.4c102.5-5.7,191.2-62.8,240.8-146c20.3-34.1,34.1-72.5,39.6-113.7c1-7.4,1.7-15,2.1-22.6
L443,160.7z\\"></path></svg><span style=\\"padding-left:45px\\">test label</span></span>"
`;

exports[`Label behaviour Markup Base with custom styles 1`] = `"<span style=\\"test:new test styles;color:#191716;background:transparent\\" class=\\"css-t2xfrr\\"><span style=\\"padding-left:45px\\">test label</span></span>"`;

exports[`Label behaviour Markup Base with icon 1`] = `"<span style=\\"color:#191716;background:transparent\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width:32px;height:32px;z-index:1;opacity:1\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>Bitcoin</title><circle style=\\"fill:#edda48\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill:#e2b255\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><path style=\\"fill:#edda48\\" d=\\"M262.5,193.9c47-38.9-0.7-74.8-36.9-73.2v-38h-26.1v38.5H186V82.8h-24.2v38h-45.1v22.5h28.6v119.9h-28.6v23.1 h45.1v37.4h23.7v-37.9h14.1v37.9h23.9v-37.9C279,284.1,319.4,238.7,262.5,193.9z M186,145.8h34.4c13.2,1.4,26.2,27.8,0,42.1H186 V145.8z M225.1,261.9H186v-49.6h39.1C252.9,220.8,254.8,249.2,225.1,261.9z\\"></path></svg><span style=\\"padding-left:45px\\">test label</span></span>"`;
